java/ql/src/semmle/code/java/frameworks/JavaxJson.qll

```diff
@@ -8,103 +8,129 @@ private import semmle.code.java.dataflow.ExternalFlow
+          ".json;JsonNumber;false;longValueExact;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonNumber;false;numberValue;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getBoolean;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getBoolean;;;Argument[1];ReturnValue;value",
+          ".json;JsonObject;false;getInt;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getInt;;;Argument[1];ReturnValue;value",
+          ".json;JsonObject;false;getJsonArray;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getJsonNumber;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getJsonObject;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getJsonString;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getString;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObject;false;getString;;;Argument[1];ReturnValue;value",
+          ".json;JsonObjectBuilder;false;add;;;Argument[-1];ReturnValue;value",
+          ".json;JsonObjectBuilder;false;add;;;Argument[1];Argument[-1];taint",
+          ".json;JsonObjectBuilder;false;addAll;;;Argument[0];ReturnValue;value",
+          ".json;JsonObjectBuilder;false;addAll;;;Argument[-1];ReturnValue;value",
+          ".json;JsonObjectBuilder;false;addNull;;;Argument[-1];ReturnValue;value",
+          ".json;JsonObjectBuilder;false;build;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonObjectBuilder;false;remove;;;Argument[-1];ReturnValue;value",
+          ".json;JsonPatch;false;apply;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonPatch;false;apply;;;Argument[0];ReturnValue;taint",
+          ".json;JsonPatch;false;toJsonArray;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;add;;;Argument[0..1];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;add;;;Argument[-1];ReturnValue;value",
+          ".json;JsonPatchBuilder;false;build;;;Argument[-1];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;copy;;;Argument[0..1];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;copy;;;Argument[-1];ReturnValue;value",
+          ".json;JsonPatchBuilder;false;move;;;Argument[0..1];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;move;;;Argument[-1];ReturnValue;value",
+          ".json;JsonPatchBuilder;false;remove;;;Argument[0];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;remove;;;Argument[-1];ReturnValue;value",
+          ".json;JsonPatchBuilder;false;replace;;;Argument[0..1];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;replace;;;Argument[-1];ReturnValue;value",
+          ".json;JsonPatchBuilder;false;test;;;Argument[0..1];ReturnValue;taint",
+          ".json;JsonPatchBuilder;false;test;;;Argument[-1];ReturnValue;value",
+          ".json;JsonPointer;false;add;;;Argument[-1];ReturnValue;taint",
```

Comment:
Is `add` really transferring taint from the receiver `JsonPointer`?
If so should that then in the same way apply to the other methods (`getValue`, `remove`, `replace`)?

Reply:
`add`: yes, because the pointer could contain an injection string, which would be incorporated into the result as a new key. `getValue`, `remove`, `replace`: no, they do not incorporate text from the pointer into the result (`replace` requires that the key is already present)

Reply:
Thanks for the clarification! Makes sense.